IPIP/0504-provider-query-parameter.md

```diff
@@ -0,0 +1,265 @@
+
+- If the `provider` parameter is included in the **query** (`?...`), it MAY be communicated to the server depending on the client parsing the parameter.
+- If the `provider` is encoded as a **fragment** (`#...`), it is only accessible to the client (browsers do not send fragments to the server).
+
+This distinction allows URI publishers to tailor behavior:
+
+- **Client-only mode:** Use a fragment (`#provider=...`) to ensure the server remains unaware of hint data. This is useful for privacy-preserving client apps or when hints are intended to guide only the client.
+- **Server-assisted mode:** Use query parameters (`?provider=...`) to allow the server to parse and act on provider hints. This may enable proxy behavior, similar to existing IPFS gateways like `ipfs.io` or `dweb.link`.
+
+Publishers of such URIs should consider the **security profile** and **trust assumptions** of their environment when deciding how to encode hints.
+
+This flexibility supports a spectrum of use cases—from fully local client-side fetch strategies to cooperative client-server resolution pipelines.
+
+## Design rationale
+
+TODO
+
+The rationale fleshes out the specification by describing what motivated
+the design and why particular design decisions were made.
+
+Provide evidence of rough consensus and working code within the community,
+and discuss important objections or concerns raised during discussion.
+
+### User benefit
+
+The proposed **Provider-Hinted URI** format aims to enable "non interactive" content-addressable retrieval by enabling smart clients to fetch bytes directly from specified providers. This allows for reduced latency, lower load on discovery systems, and improved resiliency. his brings several concrete benefits to users across different contexts:
+
+#### 🧑‍💻 Power Users & Developers
+
+- **Improved Resilience for Data Migration & Recovery**
+  - Enables seamless data exfiltration or migration from providers that do not participate in public discovery systems (e.g., not on Amino DHT or cid.contact). - Useful during outages or vendor lock-in scenarios.
+- **Custom Provider Prioritization**
+  - Developers can craft links that prefer self-hosted providers (e.g., local gateways, corporate infra) before falling back to public gateways or network discovery.
+
+#### 🚀 Performance-Sensitive Applications
+
+- **Reduced Latency on First Fetch**
+  - By embedding direct provider hints, clients can skip discovery lookups and go straight to fetching bytes—significantly lowering time to first byte, especially in cold-start scenarios.
+-	**Faster Initial Seeding**
+  - Clients like IPFS Desktop can opportunistically add hints to links they generate (e.g., using their peer ID or trusted gateway), enabling faster and more deterministic bootstrapping when others use those links.
```

Comment:
:thought_balloon: alternative approach for this use case is for IPFS Desktop to do manual provide on Amino DHT of the CID that is being shared. This would not require hardcoding anything in shared URL.

Reply:
Included this in https://github.com/ipfs/specs/pull/504/commits/3f6c1315ae037d324518b707047ead21194d04db